kernelci/build.py

```diff
@@ -362,6 +362,9 @@ def pull_tarball(kdir, url, dest_filename, retries):
         return False
     with tarfile.open(dest_filename, 'r:*') as tarball:
         tarball.extractall(kdir)
+    if delete:
+        if os.path.isfile(dest_filename):
```

Comment:
I would be surprised to find a case where the tarball wouldn't exist given that it was created just before in the same function.  If that did ever happen then letting it throw an exception would probably actually be a bit better.